src/orange/OrangeTrackView.hh

```diff
@@ -277,7 +351,9 @@ CELER_FUNCTION bool OrangeTrackView::is_on_boundary() const
  */
 CELER_FUNCTION Propagation OrangeTrackView::find_next_step()
 {
-    if (CELER_UNLIKELY(states_.boundary[thread_] == BoundaryResult::reentrant))
+    auto lsa = this->make_lsa();
+
+    if (CELER_UNLIKELY(lsa.boundary() == BoundaryResult::reentrant))
```

Comment:
I might need to think about this more.... reentrant boundaries should only be on the lowest level so that may need to be a separate flag...

Reply:
Won't the be handled by the fact that the LSA will give the boundary for the current level, which is the lowest possible level id? I am not sure myself, but I think that I need to make changes to the rest of the tracking mechanics to support multi-universe tracking before we can answer some of these tricky questions.